packages/runtime/src/createGatewayRuntime.ts

```diff
@@ -1002,6 +1004,14 @@ export function createGatewayRuntime<
     extraPlugins.push(usePropagateHeaders(config.propagateHeaders));
   }
 
+  if (config.upstreamRetry) {
+    extraPlugins.push(useUpstreamRetry(config.upstreamRetry));
+  }
+
+  if (config.upstreamTimeout) {
+    extraPlugins.push(useUpstreamTimeout(config.upstreamTimeout));
+  }
```

Comment:
We should probably invert the order of plugins.

Currently, the timeout plugin comes after the retry plugin. It means the executor with a timeout will wrap the executor with the retry.

We probably want the other way around, the retry executor should wrap the timeout executor. This way, a timed out request will be retryable.

```suggestion
  if (config.upstreamTimeout) {
    extraPlugins.push(useUpstreamTimeout(config.upstreamTimeout));
  }
  
  if (config.upstreamRetry) {
    extraPlugins.push(useUpstreamRetry(config.upstreamRetry));
  }
```